src/istgt_iscsi.c

```diff
@@ -2049,6 +2053,22 @@ istgt_iscsi_op_login(CONN_Ptr conn, ISCSI_PDU_Ptr pdu)
 			lu = NULL;
 			tsih = 0;
 
+			__sync_add_and_fetch(&discovery_counter, 1);
+
+			/*
+			 * As per the issue https://github.com/openebs/openebs/issues/2382,
+			 * 5 continous discovery requests failure can cause iscsi initiator
+			 * to get into high CPU usage.
+			 * This code checks for this state of 4 continuous discovery failures
+			 * and restarts the process.
+			 */
+			if (discovery_counter > 4) {
+				ISTGT_ERRLOG("discovery counter %lu retry limit exceeded\n",
+				    discovery_counter);
+				sleep(1);
+				abort();
+			}
```

Comment:
Couple of questions around this:

1. where is the core placed and what is the format? Will it be core.PID or something like that? 
2. how do we know, a core dump has happened?
3. how does one gain access to the cores or how does a user upload them?

Reply:
These are manual steps that need to be done by user based on the platform.

Reply:
So, the core pattern is typically something that is defined by the host. We need to understand this, or otherwise, on istgt startup make sure we archive the logs. As this problem manifests itself mostly (AFAIK) on .rpm based distros, maybe to verify what the expected pattern is would be sufficient?

On my host at least, its handled by systemd, likely its the same for those but we need to verify this. 

See, as the container will be restarted, how would we know it ever happened? Nothing I've seen either on the control plane or here would tell me that indeed this has happened.

Reply:
If I'm not wrong, nothing we can do on this with respect to this PR.